### PR TITLE
Add skip display none

### PR DESCRIPTION
**Pull request by TangRufus**
Resolve shakacode/react_on_rails#173

### CODE REVIEW

**Comment by TangRufus** [comment]
I don't know why github shows 16 commits.
Only the last few commits are mine.

**Comment by justin808** [comment]
@TangRufus The reason is that you did not sync up your remote to our latest.

Please sync up, and rebase your commits to one. Thanks.

File: app/helpers/react_on_rails_helper.rb [hunk 1]
```diff
@@ -55,19 +58,21 @@ def react_component(component_name, props = {}, options = {})
...
-                  data: {
-                    component_name: react_component_name,
-                    props: props,
-                    trace: trace(options),
-                    generator_function: generator_function(options),
-                    expect_turbolinks: turbolinks_loaded,
-                    dom_id: dom_id
-                  })
+    component_specification_tag_options = {
+      class: "js-react-on-rails-component",
```
Comment (justin808) [inline app/helpers/react_on_rails_helper.rb]: Please remove any extraneous whitespace changes.

File: app/helpers/react_on_rails_helper.rb [hunk 2]
```diff
@@ -34,6 +34,9 @@ module ReactOnRailsHelper
   #                    true, you'll still see the errors on the server.
   #    raise_on_prerender_error: <true/false> Default to false. True will raise exception on server
   #       if the JS code throws
+  #    skip_display_none: <true/false> Default to false. True will not add `style: "display:none"`
```
Comment (justin808) [inline app/helpers/react_on_rails_helper.rb]: This doesn't need to be configurable at the component level. I.e. Let's not put this in the view helper.

File: README.md [hunk 3]
```diff
@@ -189,6 +189,7 @@ This is how you actually render the React components you exposed to `window` ins
   + **trace:** set to true to print additional debugging information in the browser. Defaults to true for development, off otherwise.
   + **replay_console:** Default is true. False will disable echoing server-rendering logs to the browser. While this can make troubleshooting server rendering difficult, so long as you have the default configuration of logging_on_server set to true, you'll still see the errors on the server.
   + **raise_on_prerender_error:** Default is false. True will throw an error on the server side rendering. Your controller will have to handle the error.
+  + **skip_display_none:** Default to false. True will not add `style: "display:none"`to react_component div. Useful when your [content security policy](https://developer.mozilla.org/en-US/docs/Web/Security/CSP/CSP_policy_directives) doesn't include `style-src: 'unsafe-inline'`
```
Comment (justin808) [inline README.md]: This should move to only the server config.

File: app/helpers/react_on_rails_helper.rb (hunk 2 above)
Comment (tangrufus) [inline app/helpers/react_on_rails_helper.rb]: Then, put it in initializer only?

Comment (justin808) [inline app/helpers/react_on_rails_helper.rb]: Yes, only to override global config.

**Comment by TangRufus** [comment]
rebased & `:skip_display_none` should be set in initializer now

**Comment by justin808** [comment]
@TangRufus we're failing CI: https://travis-ci.org/shakacode/react_on_rails/builds/99351695#L9523

**Comment by TangRufus** [comment]
Passed the tests finally.

File: app/helpers/react_on_rails_helper.rb [hunk 4]
```diff
@@ -56,18 +56,18 @@ def react_component(component_name, props = {}, options = {})
...
-                  style: "display:none",
-                  data: {
-                    component_name: react_component_name,
-                    props: props,
-                    trace: trace(options),
-                    generator_function: generator_function(options),
-                    expect_turbolinks: turbolinks_loaded,
-                    dom_id: dom_id
-                  })
+       content_tag(:div,
```
Comment (justin808) [inline app/helpers/react_on_rails_helper.rb]: why the whitespace change?

**Comment by justin808** [comment]
- [ ] Need test verifying that the configuration option works
- [ ] No whitespace or other unnecessary changes like file mode changes. See the changed files tab of the PR.

Thanks! Looks great overall.

File: app/helpers/react_on_rails_helper.rb (hunk 4 above)
Comment (tangrufus) [inline app/helpers/react_on_rails_helper.rb]: it was a mistake

**Comment by TangRufus** [comment]
How do you test the initializer config?
Do you prefer a new dummy app or other measure?

**Comment by justin808** [comment]
I think you can probably use a capybara test and set the initial value at the beginning of the test.

Take a look at https://github.com/shakacode/react_on_rails/blob/master/spec/dummy/spec/features/integration_spec.rb#L100

**Comment by justin808** [comment]
@TangRufus If you can get a spec in, I'll merge this.

**Comment by robwise** [comment]
@TangRufus Easiest way would be to add it to the spec/dummy app as an integration test

**Comment by samnang** [comment]
@TangRufus I think you have one test to test the default value of the option here https://github.com/shakacode/react_on_rails/blob/master/spec%2Freact_on_rails%2Fconfiguration_spec.rb and put another test under dummy app to verify the feature is working as expected.

**Comment by justin808** [comment]
@TangRufus We can put this one in after we release 2.0 if that's OK with you. We need a test for this.

**Comment by TangRufus** [comment]
Sorry for the delay.
I am inexperienced with tests. 

Does anyone can help writing the tests?

@justin808 No problem. Merge only if you think it's ready.

**Comment by justin808** [comment]
@TangRufus Once this PR is rebased on top of master and has a few tests, we'll merge.

**Comment by aaronvb** [comment]
@TangRufus Not sure where you are with this but I can help with writing the tests.

**Comment by justin808** [comment]
@aaronvb, go ahead!

**Comment by TangRufus** [comment]
Thanks @arronvb

**Comment by aaronvb** [comment]
@TangRufus @justin808  What's the best way for me to work on the tests other than giving me access to your repo?

**Comment by justin808** [comment]
@aaronvb merge @TangRufus change onto a branch for @shakacode and continue from there.

**Comment by aaronvb** [comment]
@justin808 @TangRufus I've pushed a branch with the tests added: https://github.com/shakacode/react_on_rails/tree/TangRufus-add-skip-display-none

**Comment by justin808** [comment]
Please create a PR and let me know when it's ready for merge, @aaronvb. Thanks!! :clap:

**Comment by TangRufus** [comment]
closing this in favour of #218

**Pull request closed without merging**